### ISSUE TITLE
Question about the accuracy on S3DIS

### ISSUE THREAD

**Issue by Treemann** [mit-han-lab/pvcnn#4]
Hi synxlin,

Thanks for your contribution!

I tried to reproduce your results on S3DIS with default configurations, and there are some gaps between mine and yours.

| version | OA | mIoU |
|---|---|---|
|PVCNN (release) | 86.16 | 56.20 |
|PVCNN (reproduce) | 84.95 | 53.96 |
|PVCNN++ (release) | 87.17 | 58.43 |
|PVCNN++ (reproduce) | 85.54 | 55.90 |

I notice that you set a random seed, so is that reasonable? Should I modify the training parameters to get a better result? (The code I use was released before your commits on Jan 5, 2020)

**Comment by synxlin** [comment]
Hi, @Treemann. Could you please provide the training environment (e.g., PyTorch version, CUDA version, GPU, number of GPUs used)?
When training on S3DIS, we find that the accuracy is sensitive to training batch size. You can either train with batch size = 32 on 2 GPUs or with batch size = 16 on 1 GPUs.
I set the random seed just for evaluation reproducibility since PointNet/PointNet++ randomly sample points at the very beginning. It doesn't affect the training.

**Comment by Treemann** [comment]
Hi, @synxlin. The settings of my training environment is listed as follow.

Pytorch version: 1.3.0
CUDA version: 9.2
GPU device: 1 V100

The training batch size is set 32 as default.

I  train the two networks again with the same default settings and still there are still some gaps.

**Comment by synxlin** [comment]
Hi, @Treemann. I have experimented on one V100. For PVCNN, could you set the batch size to 16 if you only use one GPU as I mentioned in the last reply?
How did you evaluate these accuracies and IoUs? During training, the meters do not give you accurate accuracies and IoU. You have to run the command with "--evaluate" after training to get final evaluation accuracies and IoUs.

**Comment by Treemann** [comment]
Hi, @synxlin. Thanks for your suggestion.
I use the "--evaluation" option and the result seems reasonable. I will try another experiment with 1 GPU and batch_size=16.

| version | OA | mIoU |
|---|---|---|
|PVCNN (release) | 86.16 | 56.20 |
|PVCNN (reproduce) | 86.37 | 55.33 |
|PVCNN++ (release) | 87.17 | 58.43 |
|PVCNN++ (reproduce) | 87.15 | 57.54 |

**Comment by synxlin** [comment]
Your reproduced results are in line with the results in our paper (Table 4):
PVCNN is more efficient and better than 3D-UNet (54.93 mIoU);
PVCNN++ is more efficient and better than PointCNN (57.26 mIOU).
